service/lib/agama/storage/proposal_settings_conversion/from_y2storage.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "agama/storage/proposal_settings"
+require "agama/storage/volume_conversion"
+
+module Agama
+  module Storage
+    module ProposalSettingsConversion
+      # Proposal settings conversion from Y2Storage format.
+      class FromY2Storage
+        # @param settings [Y2Storage::ProposalSettings]
+        # @param config [Agama::Config]
+        def initialize(settings, config:)
+          @settings = settings
+          @config = config
+        end
+
+        # Performs the conversion from Y2Storage format.
+        #
+        # @return [Agama::Storage::ProposalSettings]
+        def convert
+          ProposalSettings.new.tap do |target|
+            boot_device_conversion(target)
+            lvm_conversion(target)
+            encryption_conversion(target)
+            space_settings_conversion(target)
+            volumes_conversion(target)
+          end
+        end
+
+      private
+
+        # @return [Y2Storage::ProposalSettings]
+        attr_reader :settings
+
+        # @return [Agama::Config]
+        attr_reader :config
+
+        # @param target [Agama::Storage::ProposalSettings]
+        def boot_device_conversion(target)
+          target.boot_device = settings.root_device
+        end
+
+        # @param target [Agama::Storage::ProposalSettings]
+        def lvm_conversion(target)
+          target.lvm.enabled = settings.lvm
+
+          # Only assign system VG devices if candidate devices contains any device different to the
+          # root device.
+          candidate_devices = settings.candidate_devices || []
+          return unless candidate_devices.reject { |d| d == settings.root_device }.any?
```

Comment:
I think this case also deserve own log line.

Reply:
I am not sure if we need to log this kind of decisions. I would prefer to log the intial settings and the converted ones. And of course, the code should explain the details.